### PR TITLE
SWARM-1440: Separate build profiles further

### PR DESCRIPTION
**Pull request by kenfinnigan**
- [x] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [x] Have you created a [JIRA](https://issues.jboss.org/browse/SWARM) and used it in the commit message?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/wildfly-swarm/wildfly-swarm/pulls) for the same issue?
- [x] Have you built the project locally prior to submission with `mvn clean install`?

-----

Motivation
----------
Reduce what's built when `swarm.product.build` is set and remove testsuites from reactor for it.

Modify dependencyManagement so that testsuite modules don't have WF dependencies included to make testsuites "cleaner" in terms of dependencies.

Modifications
-------------
Added `build-parent` to be the parent of all artifacts we build as part of WF Swarm so that it can include any dependencies, such as feature packs, that are purely required for building.

Moved `maven-enforcer-pattern-size` into root of project.

Tweaked the build profiles so that depenendencyManagement contains appropriate WF Swarm artifacts that are then made available to testsuite modules for us. Also added some common dependencies into /testsuite so that all tests can use them.

Result
------
Doesn't impact the functioning of the product, mostly impacts what's built with `-Dswarm.product.build`.

### CODE REVIEW

**Comment by mkouba** [comment]
`arquillian/arquillian/output.txt`?

**Comment by kenfinnigan** [comment]
@mkouba was an inadvertent addition to repo last week, just removing it in here for simplicity

**Comment by mkouba** [comment]
@kenfinnigan Ah, sure. I overlooked that it's deleted and not created in this PR ;-)

File: testsuite/testsuite-buildtool/pom.xml [hunk 1]
```diff
@@ -42,11 +41,11 @@
       <artifactId>arquillian</artifactId>
       <scope>test</scope>
     </dependency>
-    <dependency>
+    <!-- <dependency>
       <groupId>org.jboss.arquillian.junit</groupId>
       <artifactId>arquillian-junit-container</artifactId>
       <scope>test</scope>
-    </dependency>
+    </dependency> -->
```
Comment (Ladicek) [inline testsuite/testsuite-buildtool/pom.xml]: Maybe remove instead of commenting out? Was removed in a lot of other modules too.

**Review by Ladicek — DISMISSED** [review]
LGTM

**Review by Ladicek — APPROVED** [review]

**Comment by scottmarlow** [comment]
testsuite/testsuite-mongo needs version.mongodb.driver, which is now in build-parent/pom.xml, should I duplicate version.mongodb.driver under testsuite?  Or should the testsuite/pom.xml parent be changed to build-parent/pom.xml?

**Comment by kenfinnigan** [comment]
Sorry @scottmarlow, we can move the version back to the root, there are some other versions in the root for that reason.

Is it just that version or more?

**Comment by scottmarlow** [comment]
Yes, a few others:

testsuite-cassandra needs version.cassandra.driver.

testsuite-orientdb needs dependency version on com.orientechnologies:orientdb-core, com.orientechnologies:orientdb-graphdb, com.orientechnologies:orientdb-object, com.orientechnologies:orientdb-client, com.tinkerpop.blueprints:blueprints-core, com.googlecode.concurrentlinkedhashmap:concurrentlinkedhashmap-lru, org.jboss.resteasy:resteasy-jackson2-provider

testsuite-neo4j needs version.neo4j.driver

Should I create a new pr to move these?  Probably easier for me to verify, until we get the NoSQL Databases running in CI.

**Comment by kenfinnigan** [comment]
if you could that would be great, I tried to figure it out the best I could yesterday when rebasing my changes on top.

Sorry for the inconvenience

**Pull request merged**